### PR TITLE
Add create folder API v2

### PR DESCRIPTION
**Pull request by phuongnq**
https://github.com/craftercms/craftercms/issues/6061

### CODE REVIEW

File: src/main/resources/crafter/studio/upgrade/pipelines.yaml [hunk 1]
```diff
@@ -1044,6 +1056,12 @@ configurations:
             - type: xsltFileUpgrader
               template: crafter/studio/upgrade/4.1.x/permission-mappings/permission-mappings-config-v4.1.2.xslt
               commitDetails: Change 'Change Content Type' permission to 'change_content_type'
+        - currentVersion: 4.1.2
+          nextVersion: 4.2.0.0
```
Comment (jmendeza) [inline src/main/resources/crafter/studio/upgrade/pipelines.yaml:1060]: An extra `.0`?

File: src/main/java/org/craftercms/studio/impl/v2/service/security/internal/UserServiceInternalImpl.java [hunk 2]
```diff
@@ -55,7 +59,7 @@ public class UserServiceInternalImpl implements UserServiceInternal {
     private static final Logger logger = LoggerFactory.getLogger(UserServiceInternalImpl.class);
 
     private final UserDAO userDao;
-    private final GroupServiceInternal groupServiceInternal;
+    private ConfigurationService configurationService;
```
Comment (jmendeza) [inline src/main/java/org/craftercms/studio/impl/v2/service/security/internal/UserServiceInternalImpl.java]: Can this be final?

File: src/main/java/org/craftercms/studio/impl/v2/service/security/internal/UserServiceInternalImpl.java [hunk 3]
```diff
@@ -64,15 +68,15 @@ public class UserServiceInternalImpl implements UserServiceInternal {
     private final Cache<String, User> userCache;
     private final Zxcvbn zxcvbn;
 
-    @ConstructorProperties({"userDao", "groupServiceInternal", "studioConfiguration", "siteService", "securityService",
+    @ConstructorProperties({"userDao", "configurationService", "studioConfiguration", "siteService", "securityService",
             "accessTokenService", "retryingDatabaseOperationFacade", "userCache", "zxcvbn"})
-    public UserServiceInternalImpl(UserDAO userDao, GroupServiceInternal groupServiceInternal,
+    public UserServiceInternalImpl(UserDAO userDao , ConfigurationService configurationService,
```
Comment (jmendeza) [inline src/main/java/org/craftercms/studio/impl/v2/service/security/internal/UserServiceInternalImpl.java]: Let's remove space before the comma

File: src/main/java/org/craftercms/studio/impl/v2/utils/spring/security/messaging/StudioMessageSecurityExpressionRoot.java [hunk 4]
```diff
@@ -69,7 +69,7 @@ public boolean isSiteMember(String siteId) {
             }
```
Comment (jmendeza) [inline src/main/java/org/craftercms/studio/impl/v2/utils/spring/security/messaging/StudioMessageSecurityExpressionRoot.java]: Please take a look at recent changes in `SecurityService`. We have `isSiteMember` and `isSiteAdmin`, so we can drop dependency of this class on UserService

**Review by jmendeza — CHANGES_REQUESTED** [review]

File: src/main/java/org/craftercms/studio/impl/v2/service/security/internal/UserServiceInternalImpl.java (hunk 2 above)
Comment (phuongnq) [reply src/main/java/org/craftercms/studio/impl/v2/service/security/internal/UserServiceInternalImpl.java]: Updated to `final`

File: src/main/java/org/craftercms/studio/impl/v2/service/security/internal/UserServiceInternalImpl.java (hunk 3 above)
Comment (phuongnq) [reply src/main/java/org/craftercms/studio/impl/v2/service/security/internal/UserServiceInternalImpl.java]: updated

File: src/main/resources/crafter/studio/upgrade/pipelines.yaml (hunk 1 above)
Comment (phuongnq) [reply src/main/resources/crafter/studio/upgrade/pipelines.yaml:1060]: I noticed that we have the same convention for `site-config.xml`: https://github.com/craftercms/studio/blob/develop/src/main/resources/crafter/studio/upgrade/pipelines.yaml#L880 so I added an extra `.0`, or do we prefer to have just 3 number for `permission-mappings.xml`?

File: src/main/java/org/craftercms/studio/impl/v2/utils/spring/security/messaging/StudioMessageSecurityExpressionRoot.java (hunk 4 above)
Comment (phuongnq) [reply src/main/java/org/craftercms/studio/impl/v2/utils/spring/security/messaging/StudioMessageSecurityExpressionRoot.java]: This was fixed after merged `develop` into this branch

**Review by jmendeza — APPROVED** [review]

**Comment by phuongnq** [comment]
Close for now. Will revaluate in 5.0.0

**Pull request closed without merging**